vault/identity_store_aliases.go

```diff
@@ -296,16 +333,17 @@ func (i *IdentityStore) handleAliasUpdate(ctx context.Context, req *logical.Requ
 		if err != nil {
 			return nil, err
 		}
-		// Bail unless it's just a case change
-		if existingAlias != nil && !strings.EqualFold(existingAlias.Name, name) {
+
+		// Bail unless it's just a case change or no custom meta data change
+		if existingAlias != nil && !strings.EqualFold(existingAlias.Name, name) && reflect.DeepEqual(customMetadata, alias.CustomMetadata) {
```

Comment:
I don't think this check should be changed.  It's also going to collide with https://github.com/hashicorp/vault/pull/12473, which I'm about to merge.

Reply:
done